ui/src/components/Inventory/InventoryFormModal.js

```diff
@@ -0,0 +1,129 @@
+import Button from '@material-ui/core/Button'
+import Dialog from '@material-ui/core/Dialog'
+import DialogActions from '@material-ui/core/DialogActions'
+import DialogContent from '@material-ui/core/DialogContent'
+import DialogTitle from '@material-ui/core/DialogTitle'
+import Grid from '@material-ui/core/Grid'
+import React from 'react'
+import TextField from '../Form/TextField'
+import { Field, Form, Formik } from 'formik'
+
+class InventoryFormModal extends React.Component {
+  render() {
+      const {
+        formName,
+        handleDialog,
+        handleInventory,
+        title,
+        initialValues,
+        units
+      } = this.props
+
+      return (
+        <Dialog
+        open={this.props.isDialogOpen}
+        maxWidth='sm'
+        fullWidth={true}
+        onClose={() => { handleDialog(false) }}
+        >
+          <Formik
+            initialValues={initialValues}
+            onSubmit={values => {
+              console.log('values after submit', values)
+              values.bestBeforeDate = values.bestBeforeDate + 'T12:00:00Z'
+              handleInventory(values)
+              handleDialog(true)
+            }}>
+            {helpers =>
+              <Form
+              noValidate
+              autoComplete='off'
+              id={formName}
+              >
+                <DialogTitle id='alert-dialog-title'>
+                  {`${title} Inventory`}
+                </DialogTitle>
+                <DialogContent>
+                  <Grid container>
+                    <Grid item xs={12} sm={12}>
+                      <Field
+                        custom={{ variant: 'outlined', fullWidth: true, }}
+                        name='name'
+                        label='Name'
+                        component={TextField}
+                      />
+                      <label for='productType'>Product Type: </label>
+                      <Field
+                        custom={{ variant: 'outlined', fullWidth: true, }}
+                        name='productType'
+                        id='productType'
+                        component='select'>
+                        <option productType=''></option>
+                        <option productType='Hops'>Hops</option>
+                        <option productType='Malt'>Malt</option>
```

Comment:
productType is a dynamic field and should allow for additional productTypes to be added. Populating the dropdown will involve a little more than that and cannot be hard coded